src/TripalCultivateValidator/TripalCultivateValidatorBase.php

```diff
@@ -0,0 +1,269 @@
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getSupportedInputTypes() {
+    return $this->pluginDefinition['input_types'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function checkInputTypeSupported(string $input_type) {
+    $supported_types = $this->getSupportedInputTypes();
+
+    if (in_array($input_type, $supported_types)) {
+      return TRUE;
+    }
+    return FALSE;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateMetadata(array $form_values) {
+    $plugin_name = $this->getValidatorName();
+    throw new \Exception("Method validateMetadata() from base class called for $plugin_name. If this plugin wants to support this type of validation then they need to override it.");
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateFile(int|null $fid) {
+    $plugin_name = $this->getValidatorName();
+    throw new \Exception("Method validateFile() from base class called for $plugin_name. If this plugin wants to support this type of validation then they need to override it.");
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateRow(array $row_values) {
+    $plugin_name = $this->getValidatorName();
+    throw new \Exception("Method validateRow() from base class called for $plugin_name. If this plugin wants to support this type of validation then they need to override it.");
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateRawRow(string $raw_row) {
+    $plugin_name = $this->getValidatorName();
+    throw new \Exception("Method validateRawRow() from base class called for $plugin_name. If this plugin wants to support this type of validation then they need to override it.");
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function checkIndices($row_values, $indices) {
+
+    // Report if the indices array is empty.
+    if (!$indices) {
+      throw new \Exception(
+        'An empty indices array was provided.'
+      );
+    }
+
+    // Get the potential range by looking at $row_values.
+    $num_values = count($row_values);
+    // Count our indices array.
+    $num_indices = count($indices);
+    if ($num_indices > $num_values) {
+      throw new \Exception(
+        'Too many indices were provided (' . $num_indices . ') compared to the number of cells in the provided row (' . $num_values . ').'
+      );
+    }
+
+    // Pull out just the keys from $row_values and compare with $indices.
+    $row_keys = array_keys($row_values);
+    $result = array_diff($indices, $row_keys);
+    if ($result) {
+      $invalid_indices = implode(', ', $result);
+      throw new \Exception(
+        'One or more of the indices provided (' . $invalid_indices . ') is not valid when compared to the indices of the provided row.'
+      );
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getConfigAllowNew() {
+    $allownew = \Drupal::config('trpcultivate.settings')
+      ->get('ontology.allownew');
+
+    return $allownew;
+  }
```

Comment:
This should be removed as it's phenotypes specific. Furthermore, it should be removed from the interface and from the tests.

Then in the phenotypes repo, we will have a base class containing this method that inherits from this class.